[executor] First instructions implementation

Dimanteo opened the pull request:
Implemented ADD, SUB, ADDI, LW, SW instructions. Also fixed some warnings and typos.

Dimanteo committed:
[executor] First instructions implementation

Implemented ADD, SUB, ADDI, LW, SW instructions. Also fixed some
warnings and typos.

DragonZoom commented on src/executor/executor.cpp:
```diff
@@ -5,9 +5,13 @@
 
-void exec_ADD(Instruction &instr, Hart &hart) {} //!
+void exec_ADD(Instruction &instr, Hart &hart) {
+  hart.setReg(instr.rd, instr.rs1 + instr.rs2);
```
instr.rs1 and instr.rs2 are register index. We shoul call getReg(instr.rs1).

DragonZoom commented on src/executor/executor.cpp:
```diff
@@ -5,9 +5,13 @@
 
-void exec_ADDI(Instruction &instr, Hart &hart) {} //!
+void exec_ADDI(Instruction &instr, Hart &hart) {
+  hart.setReg(instr.rd, instr.imm + instr.rs1);
```
The same

Dimanteo committed:
[executor] Fix index instead of value usage

- Fixed bug in instruction implementation - register indexes were used in
operations instead of register values.
- Added dump of register file.
- Removed auto-gen file from commit.
